src/Compilers/CSharp/Test/Symbol/Symbols/DefaultInterfaceImplementationTests.cs

```diff
@@ -17437,5 +17431,498 @@ static void Main()
                 Diagnostic(ErrorCode.ERR_BadAccess, "P2").WithArguments("I1.P2").WithLocation(9, 12)
                 );
         }
+
+        [Fact]
+        public void NestedTypes_01()
+        {
+            var source1 =
+@"
+public interface I1
+{
+    interface T1
+    {
+        void M1();
+    }
+
+    class T2
```

Comment:
I don't expect a problem, but maybe test type nested in type.
Also consider generic type (with it's own T, or with the T from the containing interface).